### DISCUSSION TITLE
Can't upgrade clingo

### DISCUSSION THREAD

**Discussion by monty20python** [Homebrew/discussions#117] [Everyday usage] [answered]
I tried removing and reinstalling clingo but a similar message appears when I try to do that as when I try to upgrade it. I've tried --force as well with no luck. I have a feeling it has something to do with clasp and gringo being installed, but I have no idea how to fix it and was wondering if anyone had any ideas. 

~ $ brew upgrade clingo
==> Upgrading 1 outdated package:
clingo 5.4.0_2
==> Upgrading clingo -> 5.4.0_2
==> Downloading https://homebrew.bintray.com/bottles/clingo-5.4.0_2.catalina.bottle.tar.gz
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/84f6fc12ad562a91cee3984c937f708316a1e4e3b1634f7f95fe6f71f817f238--clingo-5.4.0_2.catalina.bottle.tar.gz
Error: clingo 4.5.4 is already installed
To upgrade to 5.4.0_2, run `brew upgrade clingo`.

**Comment by jonchang** [comment]
Can you post `brew config` and `brew doctor` here?

**Reply by monty20python** [reply]
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you an update run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.


Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  python@2
  camlp4

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python2.7/greenlet/greenlet.h

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@2
  numpy
  gnupg
  python@3.9
  numpy@1.16



HOMEBREW_VERSION: 2.5.8-118-g2ac5cff
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2ac5cffd0dc261b2ab922169197e8b766b7d5748
Last commit: 35 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f822fcef2b4f3e2d76fd4dbea66f1ef0e3d76524
Core tap last commit: 60 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.sOmyO1IdUU/org.macosforge.xquartz:0
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit ivybridge
Clang: 12.0 build 1200
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.0_40
macOS: 10.15.7-x86_64
CLT: 12.0.31.1
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11

**Comment by Rylan12** [comment, marked as answer]
`clasp`, `gringo`, and `clingo` all point to the same formula (which is not named `clingo`)

I think this is the same issue as https://github.com/Homebrew/homebrew-core/issues/61522

If so, `brew uninstall clasp gringo` may do the trick.

**Reply by monty20python** [reply]
That worked, my googling abilities failed me. Thanks!
